internal/process/manager.go

```diff
@@ -4,73 +4,68 @@ import (
 	"fmt"
 
 	"github.com/eagraf/habitat-new/core/state/node"
-	"github.com/eagraf/habitat-new/internal/node/controller"
 	"github.com/eagraf/habitat-new/internal/node/hdb"
 )
 
-type RunningProcess struct {
-	*node.ProcessState
+type runningProcess struct {
+	*node.Process
+	driverID string
+}
+
+type RestoreInfo struct {
+	Procs map[string]*node.Process
+	Apps  map[string]*node.AppInstallationState
 }
 
 type ProcessManager interface {
-	ListProcesses() ([]*node.ProcessState, error)
+	ListProcesses() ([]*node.Process, error)
 	StartProcess(*node.Process, *node.AppInstallation) error
 	StopProcess(extProcessID string) error
-	GetProcess(processID string) (*node.ProcessState, error)
+	// Returns process state, true if exists, otherwise nil, false to indicate non-existence
+	GetProcess(processID string) (*node.Process, bool)
+	// ProcessManager should implement Component -- specifically, restore state given a set of processes and apps
+	node.Component[RestoreInfo]
 }
 
-type BaseProcessManager struct {
+type baseProcessManager struct {
 	processDrivers map[string]Driver
-
-	processes map[string]*RunningProcess
+	processes      map[string]*runningProcess
```

Comment:
My other PR removed these. Is there actually any point in keeping these around anymore?

Ideally components don't have state if we can help it.

Reply:
Agree, maybe I just need to rebase to pick up that change?

Reply:
Hm, I realized I'm using this for tests (the `GetProcess` and `ListProcesses` functions), maybe we can remove it in a follow-up?